dts/xtensa/espressif/esp32.dtsi

```diff
@@ -161,5 +161,12 @@
 			clocks = <&rtc ESP32_SPI3_MODULE>;
 			status = "disabled";
 		};
+
+		wifi0 {
+			compatible = "espressif,esp32-wifi";
+			label = "WIFI_0";
```

Comment:
label is not necessary.
CC: @galak

Reply:
I would leave it as is, because this is what `device list` shows. I guess this is more user-friendly than what gets automatically generated.

Reply:
Is it possible to populate the wifi device without reg/ranges property? I just figure out the warning below:

`esp32.dts.pre.tmp:141.9-145.5: Warning (simple_bus_reg): /soc/wifi0: missing or empty reg/ranges property`

Reply:
> I would leave it as is, because this is what device list shows. I guess this is more user-friendly than what gets automatically generated.

As far I know the intention is remove label property from devicetree and the drop is a WIP. I noted that a discussion around this topic was started at #32608 too. At end, the important thing is NET_DEVICE_DT_INST_DEFINE that don't need label property. If PR keeps label I don't know for how long it will be there.

@sylvioalves I think everything inside **/soc** should have.

Reply:
@nandojve I guess we can keep the label for now.  
Regarding `wifi` dtsi, we have added it inside `/soc`. However, I think it could be moved to root path as `/wifi0` instead of `/soc/wifi0`. Does that make sense?

```c
wifi0: wifi0 {
	compatible = "espressif,esp32-wifi";
	label = "WIFI_0";
	status = "disabled";
};
```

Reply:
WiFi in this case is part of SoC, so in my opinion it should be in `/soc`. In root (`/`) we put things that are for example part of the PCB, like LEDs or so.

Does WiFi peripheral have some known register area that it occupies? Or is that register address confidential? :)

Reply:
AFAIK, there is no register address and this is the reason for no address in the PR. In this case, should I set as 0 or is there any other approach for that?

Reply:
Setting it to 0 seems fine to me. It will be consistent with WIFI_0 label :)

Reply:
How IDF send commands to WIFI? I mean, no register address space to direct access means it probably is on a bus, right?

Reply:
Update the dtsi file as suggest by @galak.

Reply:
> How IDF send commands to WIFI? I mean, no register address space to direct access means it probably is on a bus, right?

I am checking if there is such a thing. But as you can see below, WIFI is not a peripheral as SPI, I2C:
https://github.com/espressif/esp-idf/blob/master/components/esp32/ld/esp32.peripherals.ld

We access it over a pre-built binary and APIs.